rust_cli/src/commands/idl.rs

```diff
@@ -0,0 +1,512 @@
+    let recent_blockhash = rpc_client
+        .get_latest_blockhash()
+        .map_err(|e| anyhow!("Failed to get recent blockhash: {}", e))?;
+
+
+    //TODO: find more elegant solution to set appropriate compute unit limit
+    let transaction_to_simulate = Transaction::new_signed_with_payer(
+        &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(200000) ,ix.clone()],
+        Some(&signer.pubkey()),
+        &[signer],
+        recent_blockhash,
+    );
+
+    let simulation = rpc_client.simulate_transaction(&transaction_to_simulate)
+    .map_err(|e| anyhow!("Failed to simulate transaction: {}", e))?;
+
+    let units_consumed: u32 = simulation.value.units_consumed.unwrap_or(0) as u32;
+
+    println!("Units consumed: {}", units_consumed);
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(units_consumed + 200) ,ix],
+        Some(&signer.pubkey()),
+        &[&signer],
+        recent_blockhash,
+    );
+
+    // let mut config = RpcSendTransactionConfig::default();
+    // config.skip_preflight = true;
+
+    let signature = rpc_client
+        .send_and_confirm_transaction_with_spinner_and_commitment(&transaction, CommitmentConfig::confirmed())
+        .map_err(|e| anyhow!("Failed to send transaction: {}", e))?;
+
+    println!("Data initialized successfully!");
+    println!("Signature: {}", signature);
+    println!("Data Account: {}", idl_address);
+    
+    Ok(())
+}
+
+fn create_buffer(
+    idl_data: Vec<u8>, 
+    rpc_client: &solana_client::rpc_client::RpcClient, 
+    signer: &Keypair, 
+    priority_fees_per_cu: u64
+) -> Result<(Vec<u8>, Keypair), anyhow::Error> {
+    let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+    encoder.write_all(&idl_data)?;
+    let compressed_data = encoder.finish()?;
+    let data_len = compressed_data.len() + METADATA_OFFSET;
+    let buffer_keypair = Keypair::new();
+    let rent = rpc_client.get_minimum_balance_for_rent_exemption(data_len)
+        .map_err(|e| anyhow!("Failed to get rent: {}", e))?;
+    let create_account_ix = solana_sdk::system_instruction::create_account(
+        &signer.pubkey(),
+        &buffer_keypair.pubkey(),
+        rent,
+        data_len as u64,
+        &UPLOAD_IDL_ANCHOR_ID,
+    );
+    let create_buffer = CreateBufferBuilder::new()
+        .buffer(buffer_keypair.pubkey())
+        .authority(signer.pubkey())
+        .instruction();
+    let recent_blockhash = rpc_client
+        .get_latest_blockhash()
+        .map_err(|e| anyhow!("Failed to get recent blockhash: {}", e))?;
+    let transaction = Transaction::new_signed_with_payer(
+            &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(5000),create_account_ix, create_buffer],
```

Comment:
better not hard code this. If compute changes in the future we may run into problems. Better calculate properly remove or put at least a 2x buffer